### PR TITLE
Sasb 1761 credential export

### PR DESCRIPTION
**Pull request by marc-tranzer**

### CODE REVIEW

File: src/main/resources/com/datatheorem/mobileappsecurity/jenkins/plugin/SendBuildToDataTheoremPublisher/help-applicationCredentialPassword.html [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<div>
+    <p>
+       In order to run a full dynamic scan with an application with a login page. Data Theorem needs a test account.
```
Comment (probablyodd) [inline src/main/resources/com/datatheorem/mobileappsecurity/jenkins/plugin/SendBuildToDataTheoremPublisher/help-applicationCredentialPassword.html:3]: s/"login page. Data Theorem...."/"login page, Data Theorem..."

File: src/main/resources/com/datatheorem/mobileappsecurity/jenkins/plugin/SendBuildToDataTheoremPublisher/help-applicationCredentialUsername.html [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<div>
+    <p>
+        In order to run a full dynamic scan with an application with a login page. Data Theorem needs a test account.
```
Comment (probablyodd) [inline src/main/resources/com/datatheorem/mobileappsecurity/jenkins/plugin/SendBuildToDataTheoremPublisher/help-applicationCredentialUsername.html:3]: s/"login page. Data Theorem...."/"login page, Data Theorem..."

**Review by probablyodd — APPROVED** [review]

File: src/main/java/com/datatheorem/mobileappsecurity/jenkins/plugin/sendbuild/SendBuildAction.java [hunk 3]
```diff
@@ -430,4 +447,23 @@ private void AddContentToEntity(MultipartEntityBuilder entityBuilder, String bin
             ));
         }
     }
+
+    public String getApplicationCredentialPassword() {
+        return ApplicationCredentialPassword;
+    }
+
+    public void setApplicationCredentialPassword(String applicationCredentialPassword) {
+        listener.getLogger().println(applicationCredentialPassword);
```
Comment (superlama) [inline src/main/java/com/datatheorem/mobileappsecurity/jenkins/plugin/sendbuild/SendBuildAction.java:456]: Do we actually want to log this information?

File: src/main/java/com/datatheorem/mobileappsecurity/jenkins/plugin/sendbuild/SendBuildAction.java [hunk 4]
```diff
@@ -398,6 +400,21 @@ HttpResponse uploadBuildRequest()
         }
         listener.getLogger().println("Build file path is: " + buildPath);
 
+        // Add credential username
+        if (getApplicationCredentialUsername() != null && !getApplicationCredentialPassword().isEmpty()){
```
Comment (superlama) [inline src/main/java/com/datatheorem/mobileappsecurity/jenkins/plugin/sendbuild/SendBuildAction.java:404]: At this point, it has not been verified that `getApplicationCredentialPassword()` is not `null`; I think the expected thing would be to add an username and a password only if both are provided.

File: src/main/java/com/datatheorem/mobileappsecurity/jenkins/plugin/SendBuildToDataTheoremPublisher.java [hunk 5]
```diff
@@ -66,6 +69,8 @@ public SendBuildToDataTheoremPublisher(
         this.proxyPort = proxyPort;
         this.proxyUsername = proxyUsername;
         this.proxyPassword = Secret.fromString(proxyPassword);
+        this.applicationCredentialUsername = applicationCredentialUsername;
```
Comment (superlama) [inline src/main/java/com/datatheorem/mobileappsecurity/jenkins/plugin/SendBuildToDataTheoremPublisher.java:72]: A third field is supported, but not included here (`comments`). Also, these fields could also be grouped.

**Review by superlama — APPROVED** [review]

**Pull request merged**